src/NodeServices/Interfaces.cs

```diff
@@ -235,6 +235,7 @@ internal interface IExternalCodeCompletionProviderCore
         /// <param name="dynamoCorePath"></param>
         void Initialize(string dynamoCorePath);
 
+        void Dispose();
```

Comment:
hmmm, I think this is strange, can you try to implement `IDisposable` on the concrete types instead?
That seems much more idiomatic for c#.

Reply:
One reason I say this is that changing interfaces after they are released is usually an API break - it's harder to judge here because this is an internal interface... so likely not an issue, but I'm unsure.

Reply:
We can use IDisposable interface for DSCPythonCodeCompletionProviderCore and IronPythonCodeCompletionProviderCore but we would need the dispose() in the interface to call it on the completionProviderAdaptor in line 222 of ScriptEditorWindow.xaml.cs .

Reply:
you can do a check like this
(if provider is IDisposable dis){
dis.Dispose()
}

uses c#7 pattern match to get rid of the `as` statement.